feast_component/executor.py

```diff
@@ -16,13 +16,165 @@
+    json_format.Parse(custom_config, config)
+    s = Struct()
+    config.custom_config.Unpack(s)
+    config_dict = MessageToDict(s)
+    return config_dict
+
+
+def _load_feast_feature_store(custom_config: Dict[str, Any]) -> feast.FeatureStore:
+    with tempfile.TemporaryDirectory() as t:
+        with open(os.path.join(t, "feature_store.yaml"), "w") as f:
+            f.write(custom_config[_REPO_CONFIG_KEY])
+
+    return feast.FeatureStore(repo_path=t)
+
+
+def _get_retrieval_job(
+    entity_query: str, custom_config: Dict[str, Any]
+) -> RetrievalJob:
+    """Get feast retrieval job
+
+    Args:
+        entity_query (str): entity query.
+        custom_config (Dict[str, Any]): Custom configuration from component
+
+    Raises:
+        RuntimeError: [description]
+
+    Returns:
+        RetrievalJob: [description]
+    """
+    feature_list = custom_config[_FEATURE_KEY]
+    feature_service = custom_config[_FEATURE_SERVICE_KEY]
+    fs = _load_feast_feature_store(custom_config)
+
+    if feature_list:
+        features = feature_list
+    elif feature_service:
+        features = fs.get_feature_service(feature_service)
+    else:
+        raise RuntimeError("Either feature service or feature list should be provided")
+
+    return fs.get_historical_features(entity_df=entity_query, features=features)
 
-  def GetInputSourceToExamplePTransform(self) -> beam.PTransform:
-    """Returns PTransform for Feast to TF examples."""
+
+def _get_gcp_project(exec_properties: Dict[str, Any]) -> Optional[str]:
+    # Get GCP project from exec_properties.
+    beam_pipeline_args = exec_properties["_beam_pipeline_args"]
+    pipeline_options = beam.options.pipeline_options.PipelineOptions(beam_pipeline_args)
+    # Try to parse the GCP project ID from the beam pipeline options.
+    project = pipeline_options.view_as(
+        beam.options.pipeline_options.GoogleCloudOptions
+    ).project
+    if isinstance(project, value_provider.ValueProvider):
+        return project.get()
     return None
+
+
+@beam.ptransform_fn
+@beam.typehints.with_input_types(beam.Pipeline)
+@beam.typehints.with_output_types(bytes)
+def _FeastToExampleTransform(
+    pipeline: beam.Pipeline, exec_properties: Dict[str, Any], split_pattern: str
+) -> beam.pvalue.PCollection:
+    """Read from BigQuery and transform to TF examples.
+
+    Args:
+      pipeline: beam pipeline.
+      exec_properties: A dict of execution properties.
+      split_pattern: Split.pattern in Input config, a BigQuery sql string.
+
+    Returns:
+      PCollection of TF examples.
+    """
+    # Load custom config dictionary
+    custom_config = _load_custom_config(exec_properties["custom_config"])
+
+    # Get Feast retrieval job
+    retrieval_job = _get_retrieval_job(
+        entity_query=split_pattern, custom_config=custom_config
+    )
+
+    # Setup datasource and converter.
+    if isinstance(retrieval_job, BigQueryRetrievalJob):
+        query = retrieval_job.to_sql()
```

Comment:
Watch out for [this known discrepancy](https://github.com/feast-dev/feast/pull/1916/files#r719809575) between the `to_sql` and `to_bigquery` queries

Reply:
Seems main discrepancy is the table name used for entity_df as generated here: https://github.com/feast-dev/feast/blob/963d3ac9349b6d45a5b4a1cbd1a583655693bfa9/sdk/python/feast/infra/offline_stores/bigquery.py#L368.

This shouldn't matter  as we dont necessarily mind about the name used here.